.github/workflows/ci.yaml

```diff
@@ -0,0 +1,67 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - 'master'
+  pull_request:
+
+jobs:
+  lint:
+    name: Format and lint
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        name: Install Rust
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+          components: rustfmt, clippy
+      - uses: actions-rs/cargo@v1
+        name: Format Check
+        with:
+          command: fmt
+          args: -- --check
+      - uses: actions-rs/cargo@v1
+        name: Lint
+        with:
+          command: clippy
+          args: --all-features --all-targets
```

Comment:
I think you need a way to make sure the linting returns a non-zero status code like -

```
cargo clippy --all-targets --all-features -- -D warnings
```

actually just found this great recipe - https://github.com/actions-rs/meta/blob/master/recipes/matrix.md